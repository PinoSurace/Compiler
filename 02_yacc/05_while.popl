VAR stop? IS 0;
VAR count IS 0;
WHILE stop? = 0
DO
  Print;
  Print("----------------------");
  Print;
  count <- count + 1;
  IF count > 10 THEN
    stop? <- 1;
  ENDIF;
ENDWHILE;

WHILE aa DO
  WHILE bb DO
    WHILE cc DO
      Something;
    ENDWHILE;
  ENDWHILE;
ENDWHILE;

[ body part cannot be empty ]
WHILE Print != 7
DO
  [ empty ]
ENDWHILE;
