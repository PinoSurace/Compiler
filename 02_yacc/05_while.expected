atom( 0 )
factor
term
expr
var_definition( stop? )
atom( 0 )
factor
term
expr
var_definition( count )
atom( stop? )
factor
term
atom( 0 )
factor
term
expr
function_call
statement_seq
atom( ---------------------- )
factor
term
expr
function_call
statement_seq
function_call
statement_seq
atom( count )
factor
term
atom( 1 )
factor
term
expr
assignment( count )
statement_seq
atom( count )
factor
term
atom( 10 )
factor
term
expr
atom( 1 )
factor
term
expr
assignment( stop? )
statement_seq
if_statement
statement_seq
while_statement
statement_seq
atom( aa )
factor
term
expr
atom( bb )
factor
term
expr
atom( cc )
factor
term
expr
function_call
statement_seq
while_statement
statement_seq
while_statement
statement_seq
while_statement
statement_seq
function_call
atom
factor
term
atom( 7 )
factor
term
expr
26:Syntax Error (token: 'ENDWHILE')
